src/text/parsers/float.rs

```diff
@@ -33,8 +33,10 @@ fn float_special_value(input: &str) -> IResult<&str, TextValue> {
 fn float_numeric_value(input: &str) -> IResult<&str, TextValue> {
     map_res::<_, _, _, _, ParseFloatError, _, _>(
         recognize(tuple((
-            opt(tag("-")),
-            alt((floating_point_number, digits_before_dot)),
+            alt((
+                floating_point_number,
+                recognize(pair(opt(tag("-")), digits_before_dot)),
```

Comment:
🗺️ `floating_point_number` already accepts an optional `-`, so having `opt(tag("-"))` outside the `alt(...)` meant that `float_numeric_value` could accept _two_ optional `-` signs. 😬 Now the `opt(tag("-"))` only applies to `digits_before_dot`.